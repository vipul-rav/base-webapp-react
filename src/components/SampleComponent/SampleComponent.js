import React, { memo } from 'react';

const SampleComponent = memo(({ postList }) => (
    <div>
        A sample component for post List:
        <ul>
            {postList &&
                postList.map(post => <li key={post.id}>{post.title}</li>)}
        </ul>
    </div>
));

export { SampleComponent };
