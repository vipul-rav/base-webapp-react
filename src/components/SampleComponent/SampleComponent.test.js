import React from 'react';
import { render } from 'setupTests';
import { SampleComponent } from './SampleComponent';
import configureMockStore from 'redux-mock-store';

const content = {
    ExitText: 'Exit',
    btnContinue: 'Continue',
    homePageText: 'Home Page',
};

const setup = overridesProps => {
    const props = {
        content: content,

        ...overridesProps,
    };

    const mockStore = configureMockStore();
    const store = mockStore({
        config: {
            externalContent: content,
        },
    });

    const { wrapper, getByText } = render(
        <SampleComponent {...props} store={store} />
    );

    return {
        props,
        wrapper,
        getByText,
    };
};

describe('Sample Component', () => {
    it('should render', () => {
        const overrideProps = {
            isLoading: true,
            postList: [
                {
                    id: '1',
                    title: 'title1',
                },
            ],
        };
        const { wrapper } = setup(overrideProps);
        expect(wrapper).toMatchSnapshot();
    });
});
