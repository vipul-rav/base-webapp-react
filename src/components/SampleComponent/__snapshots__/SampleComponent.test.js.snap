// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Component should render 1`] = `undefined`;
